File: ts/wam/4780.ts
export const WamEphemeralSyncResponseReceive = 4780 as const

export interface EphemeralSyncResponseReceiveEvent {
  /** field 1, wire `client_disappearing_mode_initiator` */
  clientDisappearingModeInitiator?: DISAPPEARINGCHATINITIATORTYPE
  /** field 2, wire `client_ephemerality_duration` */
  clientEphemeralityDuration?: number
  /** field 3, wire `client_ephemerality_initiator` */
  clientEphemeralityInitiator?: EPHEMERALITYINITIATORTYPE
  /** field 4, wire `client_ephemerality_setting_timestamp` */
  clientEphemeralitySettingTimestamp?: number
  /** field 5, wire `client_ephemerality_trigger_action` */
  clientEphemeralityTriggerAction?: EPHEMERALITYTRIGGERACTIONTYPE
  /** field 6, wire `esr_disappearing_mode_initiator` */
  esrDisappearingModeInitiator?: DISAPPEARINGCHATINITIATORTYPE
  /** field 7, wire `esr_ephemerality_duration` */
  esrEphemeralityDuration?: number
  /** field 8, wire `esr_ephemerality_initiator` */
  esrEphemeralityInitiator?: EPHEMERALITYINITIATORTYPE
  /** field 9, wire `esr_ephemerality_setting_timestamp` */
  esrEphemeralitySettingTimestamp?: number
  /** field 10, wire `esr_ephemerality_trigger_action` */
  esrEphemeralityTriggerAction?: EPHEMERALITYTRIGGERACTIONTYPE
  /** field 11, wire `esr_failure_reason` */
  esrFailureReason?: ESRFAILUREREASONTYPE
  /** field 12, wire `esr_resolve_result` */
  esrResolveResult?: ESRSENDRESULTTYPE
  /** field 13, wire `is_a_group` */
  isAGroup?: boolean
  /** field 14, wire `thread_id` */
  threadId?: string
}

File: rust/wam/4780.rs
pub const WAM_EPHEMERAL_SYNC_RESPONSE_RECEIVE: u32 = 4780;

#[derive(Debug, Default)]
pub struct EphemeralSyncResponseReceiveEvent {
    /// field 1, wire `client_disappearing_mode_initiator`
    pub client_disappearing_mode_initiator: Option<DISAPPEARINGCHATINITIATORTYPE>,
    /// field 2, wire `client_ephemerality_duration`
    pub client_ephemerality_duration: Option<i64>,
    /// field 3, wire `client_ephemerality_initiator`
    pub client_ephemerality_initiator: Option<EPHEMERALITYINITIATORTYPE>,
    /// field 4, wire `client_ephemerality_setting_timestamp`
    pub client_ephemerality_setting_timestamp: Option<i64>,
    /// field 5, wire `client_ephemerality_trigger_action`
    pub client_ephemerality_trigger_action: Option<EPHEMERALITYTRIGGERACTIONTYPE>,
    /// field 6, wire `esr_disappearing_mode_initiator`
    pub esr_disappearing_mode_initiator: Option<DISAPPEARINGCHATINITIATORTYPE>,
    /// field 7, wire `esr_ephemerality_duration`
    pub esr_ephemerality_duration: Option<i64>,
    /// field 8, wire `esr_ephemerality_initiator`
    pub esr_ephemerality_initiator: Option<EPHEMERALITYINITIATORTYPE>,
    /// field 9, wire `esr_ephemerality_setting_timestamp`
    pub esr_ephemerality_setting_timestamp: Option<i64>,
    /// field 10, wire `esr_ephemerality_trigger_action`
    pub esr_ephemerality_trigger_action: Option<EPHEMERALITYTRIGGERACTIONTYPE>,
    /// field 11, wire `esr_failure_reason`
    pub esr_failure_reason: Option<ESRFAILUREREASONTYPE>,
    /// field 12, wire `esr_resolve_result`
    pub esr_resolve_result: Option<ESRSENDRESULTTYPE>,
    /// field 13, wire `is_a_group`
    pub is_agroup: Option<bool>,
    /// field 14, wire `thread_id`
    pub thread_id: Option<String>,
}